js/history.js

```diff
@@ -3,17 +3,80 @@ Fliplet.Registry.set('history-stack', (function() {
   var stack = [];
   var currentIndex = 0;
 
+  const columnsInfo = {
+    original: () => stack[0].data[0],
+    reference: [],
+    getReference: () => {
+      if (!columnsInfo.reference.length) {
+        columnsInfo.reference = columnsInfo.original();
+      }
+      return columnsInfo.reference;
+    },
+    current: [],
+    removedSinceReference: [],
+    removeColumns: (index, amount) => {
+      const reference = columnsInfo.getReference();
+      const removed = columnsInfo.reference.slice(index, index + amount);
+
+      const removedSinceReference = removed.filter(column => reference.includes(column));
+      columnsInfo.removedSinceReference = [...new Set([...columnsInfo.removedSinceReference, ...removedSinceReference])].filter(Boolean);
+
+      columnsInfo.reference.splice(index, amount);
+    },
+    moveColumnsInCollection: (collection, columnOriginalIndexes, toIndex) => {
+      const columnsToMove = columnOriginalIndexes.map(index => collection[index]);
+
+      const collectionClone = [...collection];
+      collectionClone.splice(columnOriginalIndexes[0], columnOriginalIndexes.length);
+      collectionClone.splice(toIndex, 0, ...columnsToMove);
+
+      return collectionClone;
+    },
+    moveColumns: (columnOriginalIndexes, toIndex) => {
+      const reference = columnsInfo.getReference();
+      const current = columnsInfo.current;
+
+      columnsInfo.current = columnsInfo.moveColumnsInCollection(current, columnOriginalIndexes, toIndex);
+      columnsInfo.reference = columnsInfo.moveColumnsInCollection(reference, columnOriginalIndexes, toIndex);
+    },
+    getRenamedColumns: () => {
```

Comment:
This is another internal function. Check all returned properties/methods in the column class to make sure only the methods being used are returned -- unless they complement existing properties. e.g.

```
function addColumn() {
}

function removeColumn() {
}

return {
  addColumn, // Used outside of the class
  removeColumn // Not used outside of the class, but is a fair method to be available because it's complementary to addColumn
};
```